### ISSUE TITLE
Patch: Incorrect RBD code causes failure to create RBD devices

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#368]
Originally reported of Google Code with ID 312.

```
What software version are you running? Please provide the output of "gnt-
cluster --version" and "gnt-cluster version".

gnt-cluster (ganeti v2.6.0-25-g27e15be) 2.6.1

Software version: 2.6.1
Internode protocol: 2060000
Configuration format: 2060000
OS api version: 20
Export interface: 0


<b>What distribution are you using?</b>
Ubuntu 12.04 with 3.6.3 kernel from the Ceph team (recommended for kernel RBD client).


<b>What steps will reproduce the problem?</b>
1. Attempt to create an instance using the RBD backend.


<b>What is the expected output? What do you see instead?</b>
Ganeti expects different output than is actually displayed from "rbd showmapped", leading to a failure to create a device.

It's possible that the output has changed in newer rbd versions; I'm not sure as I haven't used old versions.


<b>Please provide any additional information below.</b>
Attached is a patch that fixes two problems:

1) When the output of "rbd showmapped" was empty (due to no device having been created) an error was thrown. In fact, this is exactly what is expected if no device has yet been created, and as such None should be returned so that creation proceeds.

2) The output uses spaces, in order for the column fields to line up. The split filter was using tabs, however. Removing the tab filter and defaulting to the normal Python split behavior (split on runs of whitespace) allows th parsing to behave correctly.
```
Originally added on 2012-11-15 22:59:53 +0000 UTC.

Attachments: 
[bdev.py.diff](https://storage.googleapis.com/google-code-attachments/ganeti/issue-312/comment-0/bdev.py.diff)

**Comment by Ganeti-Issues-Migrator** [comment]
```
Which version of rbd tool are you using?

When the patch was implemented "rbd showmapped" returned just the headers, if no devices were mapped. So you got this:

root@node1:/# rbd showmapped
id	pool	image	snap	device

We explicitely had that check there, to ensure that the format hasn't changed.
So maybe indeed the format has changed (judging from your comment with the seperator too). If so, we need to adjust the _ParseRbdShowmappedOutput. I'll probable send an email to the Ceph list for this

```
Originally added on 2012-11-16 10:17:53 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Hmm, if the output of rdp changes between versions, it would be very good if we can add test data files with version-specific output. Look, for example, at test/data:

├── proc_drbd80-emptyline.txt
├── proc_drbd83_sync_krnl2.6.39.txt
├── proc_drbd83_sync.txt
├── proc_drbd83.txt
├── proc_drbd8.txt
├── vgreduce-removemissing-2.02.02.txt
├── vgreduce-removemissing-2.02.66-fail.txt
├── vgreduce-removemissing-2.02.66-ok.txt
├── vgs-missing-pvs-2.02.02.txt
└── vgs-missing-pvs-2.02.66.txt

So that we actually have unittests which record these differences. Thanks!
```
Originally added on 2012-11-16 12:12:45 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
I'm using the current testing release 0.54 -- this is very close to the next stable release, which will be 0.55.

The rbd tool shows *no* output when there are no devices mapped; also the fields are space-separated (with multiple spaces to make the output line up) rather than tab-separated.
```
Originally added on 2012-11-16 12:45:07 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit bdecfea26196996c4467dc28fc82a43a75de6dd8
Author: Stratos Psomadakis <psomas@grnet.gr>
Date:   Fri Feb 8 18:36:34 2013 +0200

    Fix rbd showmapped output parsing
    
    'rbd showmapped' output formatting differs between older and newer versions of
    the ceph tools. Try to use json output formatting if available (currently
    available only in the ceph master branch). For bobtail, argonaut and older
    releases fallback to manually parsing the 'rbd showmapped' output, handling the
    differences in the output format for each rbd version correctly.
    
    Signed-off-by: Stratos Psomadakis <psomas@grnet.gr>
    Reviewed-by: Guido Trotter <ultrotter@google.com>

```
Originally added on 2013-03-12 16:55:56 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released in the 2.7 series.
```
Originally added on 2013-05-09 05:45:16 +0000 UTC.

Changed State: Released
Added to Milestone: Release2.7
